src/nvim/option_defs.h

```diff
@@ -209,44 +209,58 @@
-#define STL_PAGENUM     'N'             /* page number (when printing)*/
-#define STL_VIM_EXPR    '{'             /* start of expression to substitute */
-#define STL_MIDDLEMARK  '='             /* separation between left and right */
-#define STL_TRUNCMARK   '<'             /* truncation mark if line is too long*/
-#define STL_USER_HL     '*'             /* highlight from (User)1..9 or 0 */
-#define STL_HIGHLIGHT   '#'             /* highlight name */
-#define STL_TABPAGENR   'T'             /* tab page label nr */
-#define STL_TABCLOSENR  'X'             /* tab page close nr */
-#define STL_ALL         ((char_u *) "fFtcvVlLknoObBrRhHmYyWwMqpPaN{#")
+/// 'statusline' option flags
+enum {
+  STL_FILEPATH        = 'f',  ///< Path of file in buffer.
+  STL_FULLPATH        = 'F',  ///< Full path of file in buffer.
+  STL_FILENAME        = 't',  ///< Last part (tail) of file path.
+  STL_COLUMN          = 'c',  ///< Column og cursor.
+  STL_VIRTCOL         = 'v',  ///< Virtual column.
+  STL_VIRTCOL_ALT     = 'V',  ///< - with 'if different' display.
+  STL_LINE            = 'l',  ///< Line number of cursor.
+  STL_NUMLINES        = 'L',  ///< Number of lines in buffer.
+  STL_BUFNO           = 'n',  ///< Current buffer number.
+  STL_KEYMAP          = 'k',  ///< 'keymap' when active.
+  STL_OFFSET          = 'o',  ///< Offset of character under cursor.
+  STL_OFFSET_X        = 'O',  ///< - in hexadecimal.
+  STL_BYTEVAL         = 'b',  ///< Byte value of character.
+  STL_BYTEVAL_X       = 'B',  ///< - in hexadecimal.
+  STL_ROFLAG          = 'r',  ///< Readonly flag.
+  STL_ROFLAG_ALT      = 'R',  ///< - other display.
+  STL_HELPFLAG        = 'h',  ///< Window is showing a help file.
+  STL_HELPFLAG_ALT    = 'H',  ///< - other display.
+  STL_FILETYPE        = 'y',  ///< 'filetype'.
+  STL_FILETYPE_ALT    = 'Y',  ///< - other display.
+  STL_PREVIEWFLAG     = 'w',  ///< Window is showing the preview buf.
+  STL_PREVIEWFLAG_ALT = 'W',  ///< - other display.
+  STL_MODIFIED        = 'm',  ///< Modified flag.
+  STL_MODIFIED_ALT    = 'M',  ///< - other display.
+  STL_QUICKFIX        = 'q',  ///< Quickfix window description.
+  STL_PERCENTAGE      = 'p',  ///< Percentage through file.
+  STL_ALTPERCENT      = 'P',  ///< Percentage as TOP BOT ALL or NN%.
+  STL_ARGLISTSTAT     = 'a',  ///< Argument list status as (x of y).
+  STL_PAGENUM         = 'N',  ///< Page number (when printing).
+  STL_VIM_EXPR        = '{',  ///< Start of expression to substitute.
+  STL_MIDDLEMARK      = '=',  ///< Separation between left and right.
+  STL_TRUNCMARK       = '<',  ///< Truncation mark if line is too long.
+  STL_USER_HL         = '*',  ///< Highlight from (User)1..9 or 0.
+  STL_HIGHLIGHT       = '#',  ///< Highlight name.
+  STL_TABPAGENR       = 'T',  ///< Tab page label nr.
+  STL_TABCLOSENR      = 'X',  ///< Tab page close nr.
+  STL_CLICK_FUNC      = '@',  ///< Click region start.
```

Comment:
I always write what can be considered sentences (absence of action is OK in Russian). Clint does not require this, but I omit a dot only in summaries (for them being taken as “topics”), and write dots otherwise.